File: app/src/main/java/com/gaffyhoares/data/di/AppComponent.java
package com.gaffyhoares.data.di;

import com.gaffyhoares.MyApplication;
import com.gaffyhoares.ui.movies.HomeActivity;
import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {AppModule.class,ViewModelFactoryModule.class,})
public interface AppComponent {

    void inject(MyApplication myApplication);

    void inject(HomeActivity activity);
}

File: app/src/main/java/com/gaffyhoares/data/di/AppComponent.kt
package com.gaffyhoares.data.di

import com.gaffyhoares.MyApplication
import com.gaffyhoares.ui.movies.HomeActivity
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [AppModule::class, ViewModelFactoryModule::class])
interface AppComponent {
    fun inject(myApplication: MyApplication)
    fun inject(activity: HomeActivity)
}